### PR TITLE
fix(sha3): Incorrect msg_ready_o timing

### PR DESCRIPTION
**Pull request by eunchan**
As reported in https://github.com/lowRISC/opentitan/pull/13921#discussion_r935911685 and https://github.com/lowRISC/opentitan/issues/14090 , SHA3 asserted `msg_ready_o` when it moves to `StPad` state even after `process_i` has been asserted.

The behavior may cause incorrect process (the behavior is checked by an assumption though)

### CODE REVIEW

**Comment by tjaychen** [comment]
@martin-lueker @mwbranstad @senelson7 
i'm not super familiar with the `entropy_src` dv scoreboard, but in general, do we try to model the condition to ensure that it produces the expected result?

**Comment by mwbranstad** [comment]
@tjaychen i will let @martin-lueker definitely answer this one, but at one time i believe the answer was yes.

File: hw/ip/kmac/rtl/sha3pad.sv
Comment (mwbranstad) [inline hw/ip/kmac/rtl/sha3pad.sv:388]: Going solely off of the pr description, this appears to be the proper fix. @martin-lueker has been looking at this in much closer detail, so I will add him as a reviewer.

**Review by mwbranstad — COMMENTED** [review]
LGTM, but will let others approve.

File: hw/ip/kmac/rtl/sha3pad.sv
Comment (tjaychen) [reply hw/ip/kmac/rtl/sha3pad.sv:388]: > @tjaychen i will let @martin-lueker definitely answer this one, but at one time i believe the answer was yes.

thanks mark. i just wanted to bring it up because in a way, entropy_src's usage of sha3 is more intensive than other modules. So it would be good to use this as a way to crosscheck for any other errors.

**Review by vogelpi — APPROVED** [review]
This looks good to me, thanks for taking care of this @eunchan !

I agree that also @martin-lueker should approve this before merging.

File: hw/ip/kmac/rtl/sha3pad.sv
Comment (martin-lueker) [reply hw/ip/kmac/rtl/sha3pad.sv:388]: I agree.
 
Yes, the latest FW_OV tests for `entropy_src` are particularly abusive of it (which is how we identified this).

**Review by martin-lueker — APPROVED** [review]

**Pull request merged**
